07-reproj.qmd

```diff
@@ -175,9 +184,106 @@ axes[0].set_title('Original (EPSG:26912)')
 axes[1].set_title('Reprojected (EPSG:4326)');
 ```
 
-Bilinear `srtm.tif` example...
+The above example, where we automatically calculate the destination grid, is appropriate when there are no specific requirements for the raster spatial properties, but rather we just wish to preserve the raster values as much as possible (this the extent and resolution are as similar as possible to the source). 
+In other situations, however, we need to reproject a raster into a specific "template", so that it corresponds, for instance, with other rasters we use in the analysis. 
+In the following code section, we reproject the `nlcd.tif` raster, again, buit this time using the `nlcd_4326.tif` reprojection result as the "template" to demonstrate this alternative workflow.
+
+First, we create a connection to our "template" raster to read its metadata:
+
+```{python}
+template = rasterio.open('output/nlcd_4326.tif')
+template.meta
+```
+
+Then, we create a write-mode connection to our destination raster, using this metadata, meaning that as the resampling result is going to have identical metadata as the "template":
```

Comment:
Seems strange that it's not in write mode automatically, imagine there's a good reason for that...

Reply:
The default for `rasterio.open` is read-mode (`'r'`). To write into a raster file we need to "open" it (if it doesn't exist = create it) in write (`w`) mode . It's similar in concept to the low-level text writing functions in Python (https://geobgu.xyz/py/conditionals_loops.html#file-object-and-reading-lines), where we open a connection either in read or write mode, then either read or write to/from it. It's going to be more natural for readers who work with "basic" Python, and less natural for readers used to R or even `geopandas`. Perhaps we can add a note to explain the background of `open` functions in Python.